### DISCUSSION TITLE
Can't get the process.env.NEXT_PUBLIC's on my AWS environments by client side

### DISCUSSION THREAD

**Discussion by mycatdoitbetter** [vercel/next.js#24154] [Help]
### What version of Next.js are you using?

10.0.8

### What version of Node.js are you using?

14.16.0

### What browser are you using?

Tested on Chrome and Firefox

### What operating system are you using?

tested on macOS Catalina and Bigsur

### How are you deploying your application?

next build and next start, at AWS Elasticbenstalk

### Describe the Bug

I'm deploying my application at AWS with elasticbenstalk to manage and set my environments variables. On localhost, it's works fine, i can access all my process.env.NEXT_PUBLIC on client side and my others (not public) on my server side (GetServerSide, etc).

But, when i'm deploying my application to aws, setting the envs on the Elasticbenstalk, i can only access the process.env (both, public and not public) by the server side.

I'm testing this by logging the all the process.env on the client side and on the server side. The envs only appers at the server side.

### Expected Behavior

I expect to get the process.env.NEXT_PUBLIC's on my client side, to, for example, send to the google analytics.

### To Reproduce

Create a env with the elasticbenstalk, set the env variables on the configurations of the elasticbenstalk on the aws console.

Try to log the envs on server side and in the client side.

**Comment by matteo-naif** [comment]
I had a similar issue and I resolved as follow: expose the environment variable throw the nextConfig.

```
 const nextConfig = {
  env: {
    NEXTAUTH_SECRET: process.env.NEXTAUTH_SECRET,
    NEXTAUTH_URL: process.env.NEXTAUTH_URL
  }
};
```
